smalldata_tools/ana_funcs/azimuthalBinning.py

```diff
@@ -97,13 +97,18 @@ def setFromFunc(self, func=None):
         if func is None:
             self._setup()
             return
-        print('set params from func ', func.__dict__.keys())
         if func._x is not None: self.x = func._x.flatten()/1e3
         if func._y is not None: self.y = func._y.flatten()/1e3
-        if func._mask is not None and isinstance(func, ROIFunc): 
-            self._mask = func._mask.astype(bool).flatten()
-        else:
-            self._mask = (~(func._mask.astype(bool))).flatten()
+        maskattr='_mask'
+        if not hasattr(func, maskattr):
+            maskattr='mask'
+            if not hasattr(func, maskattr):
+                maskattr=None
+        if maskattr is not None and getattr(func,maskattr) is not None:
+            if isinstance(func, ROIFunc): 
+                self._mask = getattr(func, maskattr).astype(bool).flatten()
+            else:
+                self._mask = (~(getattr(func,maskattr).astype(bool))).flatten()
```

Comment:
Are masks really defined with a different convention for ROIFunc compared to all other functions? IF yes we should probably make that uniform...

Reply:
I basically agree with this: I had found the inversion already there and essentially extended it. In the end, I believe this boils down to the fact that masks as used for masked arrays usually are '1' for to-be-masked entries while many other pieces of code returns masks to be used multiplicatively (including I believe the psana.Detector functions). This was fairly hard to get correct through all the chains, so proceed with care and many tests if you want to take this one.